modules/rgbd/test/test_normal.cpp

```diff
@@ -129,6 +130,40 @@ float cy = H / 2.f + 0.5f;
 Mat K = (Mat_<double>(3, 3) << focal_length, 0, cx, 0, focal_length, cy, 0, 0, 1);
 Mat Kinv = K.inv();
 
+void points3dToDepth16U(const Mat_<Vec3f>& points3d, Mat& depthMap);
+
+void points3dToDepth16U(const Mat_<Vec3f>& points3d, Mat& depthMap)
+{
+  std::vector<Point3f> points3dvec;
+  for(int i = 0; i < H; i++)
+    for(int j = 0; j < W; j++)
+      points3dvec.push_back(Point3f(points3d(i,j)[0], points3d(i,j)[1], points3d(i,j)[2]));
+    
+  std::vector<Point2f> img_points;
+  depthMap = Mat::zeros(H, W, CV_32F);
+  Vec3f R(0.0,0.0,0.0);
+  Vec3f T(0.0,0.0,0.0);
+  cv::projectPoints(points3dvec, R, T, K, Mat(), img_points);
+  
+  int index = 0;
+  for(int i = 0; i < H; i++)
+  {
+    
+    for(int j = 0; j < W; j++)
+    {
+      float value = (points3d.at<Vec3f>(i, j))[2]; // value is the z
+      depthMap.at<float>(cvRound(img_points[index].y), cvRound(img_points[index].x)) = value;
+      index++;
+    }
+  }
+  depthMap = depthMap*1000; // convert to millimeters
+  
+  Mat depthMap16U;
+  depthMap.convertTo(depthMap16U, CV_16U);
+  depthMap16U.copyTo(depthMap);
```

Comment:
yes is useless, but if I'm not wrong `convertTo` doesn't allow in-place computation